parcels/codegenerator.py

```diff
@@ -864,42 +865,43 @@ def __init__(self, fieldset, ptype=None):
     def generate(self, funcname, field_args, const_args, kernel_ast, c_include):
         ccode = []
 
+        pname = self.ptype.name + 'p'
+
         # Add include for Parcels and math header
         ccode += [str(c.Include("parcels.h", system=False))]
         ccode += [str(c.Include("math.h", system=False))]
         ccode += [str(c.Assign('double _next_dt', '0'))]
         ccode += [str(c.Assign('size_t _next_dt_set', '0'))]
 
-        # Generate type definition for particle type
-        vdecl = []
-        for v in self.ptype.variables:
-            if v.dtype == np.uint64:
-                vdecl.append(c.Pointer(c.POD(np.void, v.name)))
-            else:
-                vdecl.append(c.POD(v.dtype, v.name))
-
+        # Generate type definition for particle struct type
+        vdeclp = [c.Pointer(c.POD(v.dtype, v.name)) for v in self.ptype.variables]
```

Comment:
why is this not `vdeclp = [c.Pointer(c.POD(np.void, v.name)) for v in self.ptype.variables]` ? May that be the issue for Windows failing the JIT tests ? Because on Windows the test that fail are exclusively the JIT tests, it would be a start to look for the error in the codegenerator.

Reply:
I've just checked and changing this line to `vdeclp = [c.Pointer(c.POD(np.void, v.name)) for v in self.ptype.variables]` actually causes all linux JIT tests to fail; so is not the solution

Reply:
okay, I understand. Unfortunate - it's also annoying that the Windows log doesn't really provide a lot of information why all that failed ... @angus-g : do you have a Windows machine running this PR at your workplace ? We here only have MacOS and Linux machines for testing, and if you could provide us with a more detailed error log, we can probably fix it. Otherwise I need to install Windows on some VirtualBox machine here or so, and try to get an error log that way.

Reply:
It looks like cgen doesn't think the Windows CI is running 64-bit. Reproducing locally, I get the exception in cgen's `dtype_to_ctype`, which will only map `uint64` if `is_64_bit_platform()`. This runs:

```python
import struct as _struct

def is_64_bit_platform():
    return _struct.calcsize('l') == 8
```

Which I think is an incorrect way to test this! `platform.architecture()` gives `('64bit', 'WindowsPE')` as we'd expect.

I'll raise this upstream with cgen.

Reply:
Thanks for exploring this, @angus-g! Any way we could work around the `cgen` Issue for now, while we wait for them to fix this and release a new version?

Reply:
As a stop-gap, we could either change the cgen dependency to a different conda channel that incorporates the fix in `environment_py3_win.yml`;

or monkey-patch cgen by modifying `cgen.dtype_to_ctype` when we import it:

```python
import cgen as c

def patched_dtype_to_ctype(dtype):
  ...

c.dtype_to_ctype = patched_dtype_to_ctype
```

Neither solution is particularly clean, but at least it's possible that we'd have meaningful Windows CI